restli-tools/src/main/resources/apiVmTemplates/action.vm

```diff
@@ -15,138 +15,127 @@ Copyright (c) 2021 LinkedIn Corp.
 *#
 #if($is_interface)
   #foreach($method in $spec.actions)
-    #set($actionParamClassName = "${util.nameCapsCase($method.name)}ActionParameters")
-    #set($actionOptionalParamClassName = "${util.nameCapsCase($method.name)}ActionOptionalParameters")
-    #doc($method.schema.doc)
+    #foreach($withEG in [true, false])
+      #set($actionParamClassName = "${util.nameCapsCase($method.name)}ActionParameters")
+      #set($actionOptionalParamClassName = "${util.nameCapsCase($method.name)}ActionOptionalParameters")
+      #doc($method.schema.doc)
 
-    #if(${method.hasRequiredParams()}) ## action with no required Params will only have one API
-    @SuppressWarnings("unchecked")
-    public CompletionStage<${method.valueClassDisplayName}> ${util.nameCamelCase(${method.name})}(
-      #if(${method.isEntityAction()})
-        $spec.keyClassName $spec.idName,
-      #end
-      #foreach($param in $method.getRequiredParameters())
-        $param.paramClassDisplayName $param.paramName #if($foreach.hasNext || $method.hasOptionalParams()),#end
-      #end
-      #if($method.hasOptionalParams())
-        Function<$actionOptionalParamClassName, $actionOptionalParamClassName> optionalParamsProvider
-      #end
-    );
-    #end
-
-    @SuppressWarnings("unchecked")
-    public CompletionStage<${method.valueClassDisplayName}> ${util.nameCamelCase(${method.name})}(
-      #if(${method.isEntityAction()})
-      $spec.keyClassName $spec.idName #if(${method.hasActionParams()}),#end
-      #end
-      #if(${method.hasActionParams()})
-      Function<$actionParamClassName, $actionParamClassName> paramsProvider
-      #end
-    );
+      #if(${method.hasRequiredParams()}) ## action with no required Params will only have one API
+        #if($method.hasOptionalParams()) ## when have optional params, the optionalParamsProvider is optional
```

Comment:
I added this so it is now consistent with other apis: There is an additional API that does not need the optionalParam as argument